winch/codegen/src/isa/x64/masm.rs

```diff
@@ -108,36 +120,29 @@ impl Masm for MacroAssembler {
             );
         };
 
-        match size {
-            OperandSize::S32 => {
-                self.asm.addl(src, dst);
-            }
-            OperandSize::S64 => {
-                self.asm.add(src, dst);
-            }
-        }
+        self.asm.add(src, dst, size);
     }
 
     fn epilogue(&mut self, locals_size: u32) {
         let rsp = rsp();
         if locals_size > 0 {
-            self.asm.add_ir(locals_size as i32, rsp);
+            self.asm.add_ir(locals_size as i32, rsp, OperandSize::S64);
         }
         self.asm.pop_r(rbp());
         self.asm.ret();
     }
 
-    fn finalize(&mut self) -> &[String] {
+    fn finalize(self) -> MachBufferFinalized<Final> {
         self.asm.finalize()
     }
 }
 
 impl MacroAssembler {
-    /// Crate a x64 MacroAssembler
-    pub fn new() -> Self {
+    /// Crate a x64 MacroAssembler.
```

Comment:
s/Crate/Create/ (and s/a x64/an x64/ if I'm being pedantic, though not too important!)

Reply:
Fixed!